src/common/snippets/src/generator.cpp

```diff
@@ -51,9 +51,11 @@ LoweringResult Generator::generate(const lowered::LinearIRPtr& linear_ir, const
     }
     result.compiled_snippet = target->get_snippet();
     result.kernel_executor_table = target->get_runtime_configurator()->get_kernel_executor_table();
-    // Some kernel executors might've been registered during code emission.
+    // In static case some kernel executors might've been registered during code emission.
     // We need to update them, so appropriate kernels will be compiled.
-    result.kernel_executor_table->update_state(linear_ir);
+    // In dynamic case it should be handled by RuntimeConfigurator
+    if (!linear_ir->is_dynamic())
+        result.kernel_executor_table->update_state(linear_ir);
```

Comment:
In dynamic case, some configs of kernel executors should be updated by `RuntimeConfigurator` firstly. And only then kernels can be compiled.

Reply:
> So do you think we can make the RuntimeConfigurator handle the static case as well?

We already do it but before JIT compilation. It's made so because also `RuntimeConfigurator` calculates `master_shape` and `data_offsets` which [are used](https://github.com/openvinotoolkit/openvino/blob/c6ab24a6d7f39cba2dada53dd0f7cb1810290e32/src/plugins/intel_cpu/src/nodes/subgraph.cpp#L782) in code generation.
Moreovoer, I think it's good decision to call `RuntimeConfigurator` before codegen in static pipeline because in the future we will be able to apply the optimization `SplitDimensionM` using `RuntimeConfigurator` -> `LoopEnd` expressions will be updated and the corresponding emitters will have the updated parameters.
I'm not sure but seems like that in general each jit emitter with kernel executor table can check: if it can compile kernel now, let's generate it. Then we don't need to manually call `result.kernel_executor_table->update_state(linear_ir);` in static case 🤔

Reply:
We can also create a convention that kernel executor emitters should be registered only in constructors.
Then we can we create emitters in the RuntimeConfig for the static case and call `update_state` there. Then trigger code emission in the generator. What do you think, is it feasible?
My aim here is to restrict kernel executor table management (ideally) to one class - RuntimeConfig.

No changes for this PR, but we need to understand how to create a more universal static/dynamic design. We discussed it earlier, but this discrepancy between static and dynamic code generation still bothering me. We should at least keep all this `if (linear_ir->is_dynamic())` in one place.

Reply:
todo: update 148644